### PR TITLE
Toasts 🍞

### PR DESCRIPTION
**Pull request by gustafguner**
A neat little addition to the app. I added this to a successful save of my info to demonstrate it. We could implement it in more places.

Skärmavbild 2019-07-12 kl  14 52 29

The toasts are quite customizable 💅 
- Symbol – Either a `String` (preferably an emoji) or an `Asset`
- Body – `String`
- Text color – `UIColor` (default: `offBlack`)
- Background color – `UIColor` (default: `white`)
- Duration – `TimeInterval` (default: `5.0`)

So for example

```swift
UIApplication.shared.appDelegate.createToast(
    symbol: .character("🤖"),
    body: "Hello world!",
    textColor: .white,
    backgroundColor: .turquoise,
    duration: 10.0
)
```

Would create this toast

Skärmavbild 2019-07-12 kl  15 12 11

### CODE REVIEW

File: Src/AppDelegate.swift [hunk 1]
```diff
@@ -50,6 +68,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         ).onValue { _ in
             let loggedIn = LoggedIn()
             self.bag += self.window.present(loggedIn, options: [], animated: true)
+            
+            let toasts = Toasts(toastSignal: self.toastSignal)
```
Comment (axelbacklund) [inline Src/AppDelegate.swift]: Maybe move everything here to its own method to make it a bit cleaner?

File: Src/Toasts.swift [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                make.width.equalTo(20)
+            }
+            
+            return symbol
+        }
+        
+        return UIView()
+    }
+    
+    func getContainerEdgeInsets() -> UIEdgeInsets {
```
Comment (Iamsamwhoami) [inline Src/Toasts.swift]: maybe a computed property instead?

File: Src/Toasts.swift [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import Flow
+import Form
+import UIKit
+
+enum ToastHideDirection {
+    case left
+    case right
+}
+
+enum ToastSymbol {
```
Comment (Iamsamwhoami) [inline Src/Toasts.swift]: You could probably use an `Either<ImageAsset, Character>` instead

File: Src/Generic/Extensions/UIView/UIView+Viewable.swift [hunk 4]
```diff
@@ -142,16 +142,17 @@ extension UIView {
 extension UIStackView {
     // swiftlint:disable large_tuple
     private func materializeArrangedViewable<V: Viewable, MatterView: UIView>(
-        viewable: V
+        viewable: V,
+        atIndex: Int?
```
Comment (Iamsamwhoami) [inline Src/Generic/Extensions/UIView/UIView+Viewable.swift]: As this is not used remove it maybe?

Comment (gustafguner) [reply Src/Generic/Extensions/UIView/UIView+Viewable.swift]: Yeah, not using anymore 👍

File: Src/Toasts.swift (hunk 3 above)
Comment (gustafguner) [reply Src/Toasts.swift]: Neat!

**Pull request merged**